### PR TITLE
only trigger docs when PR approved

### PR DESCRIPTION
**Pull request by Licini**
This should make the docs workflow only triggered by the approval of PR

### CODE REVIEW

**Review by chenkasirer — APPROVED** [review]
LGTM!

**Comment by Licini** [comment]
@chenkasirer Just realised i got it wrong, it would not run on push to main and version releases, could you check and approve again to see if it works properly this time? thx!

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -6,12 +6,12 @@ on:
...
       - 'v*'
-  pull_request:
-    branches:
-      - main
+  pull_request_review:
+    types: [submitted]
 
 jobs:
   docs:
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') || (github.event_name == 'pull_request_review' && github.event.review.state == 'approved')
```
Comment (chenkasirer) [inline .github/workflows/docs.yml:14]: so when merging to `main` does it still count as a `pull_request_review` event?

Comment (Licini) [reply .github/workflows/docs.yml:14]: nop, this condition now means
`push to main` or `pushed a tag with v*` or `a pull_request_review event and it is an approval`

Comment (Licini) [reply .github/workflows/docs.yml:14]: so it should trigger when satisfy any of the three

**Review by chenkasirer — APPROVED** [review]
LGTM

File: .github/workflows/docs.yml (hunk 1 above)
Comment (chenkasirer) [reply .github/workflows/docs.yml:14]: ah right! nice, seems to have worked :)

**Pull request merged**